src/transcoding/transcode/hwaccels/vaapi.c

```diff
@@ -764,37 +757,29 @@ int
 vaapi_encode_setup_context(AVCodecContext *avctx)
 {
     TVHContext *ctx = avctx->opaque;
-    TVHVAContext *self = NULL;
     int ret = 0;
 
-    if (!(self = calloc(1, sizeof(TVHVAContext)))) {
-        tvherror(LS_VAAPI, "Encode: Failed to allocate VAAPI context (TVHVAContext)");
-        return AVERROR(ENOMEM);
-    }
-
     /* Open VAAPI device and create an AVHWDeviceContext for it*/
-    if ((ret = av_hwdevice_ctx_create(&self->hw_frame_ref, AV_HWDEVICE_TYPE_VAAPI, NULL, NULL, 0)) < 0) {
+    if ((ret = av_hwdevice_ctx_create(&ctx->hw_device_octx, AV_HWDEVICE_TYPE_VAAPI, NULL, NULL, 0)) < 0) {
         tvherror(LS_VAAPI, "Encode: Failed to open VAAPI device and create an AVHWDeviceContext for it."
                 "Error code: %s",av_err2str(ret));
-        free(self);
-        self = NULL;
-        return ret;
+        goto vaapi_encode_setup_context_exit;
```

Comment:
Goto is just "return ret"? Don't use goto then. Unless the code gets very very very hard to understand without goto, you should never use it.